### ISSUE TITLE
Consider adding user history

### ISSUE THREAD

**Issue by cmutel** [cauldron/bw_matchbox#143]
@lilliputten Can we add a pull out panel for the processes look at by the users? This would only need to visible on the "index" page, and reuse the comments pull out panel, but just list the processes they looked at last?

It would be nice if we could store this in a cookie and only access it via Javascript, but it can touch the backend if needed.

(I know that people can use their browser history, but not all our users will think of that).

**Comment by lilliputten** [comment]
Do we need to set a limit for stored items? Eg, to store only n (10, 50?) last ones? Or to store items for the last n (2, 10?) days?

**Comment by cmutel** [comment]
Honestly, just pick a number. I would say last ten processes? And no time limit.

**Comment by lilliputten** [comment]
I store the cookie now in this format:

```
recent-processes=[{"id":30474,"name":"fungicides, at regional storehouse","timestamp":1700474963381},{"id":29852,"name":"electricity, medium voltage, production from hydro power, run-of-river RER, at grid","timestamp":1700474880306}]
```

The names here are required to construct a processes list.

But: m.b. it'd better to store only identifiers and return names for tem in the `index` page?

Because:

1. It could help to save limited volume for cookies (those names are really long sometimes).
2. It could help in case if process has renamed.

But, of course, we can work this way (with names in cookie) too.

**Comment by cmutel** [comment]
https://github.com/cauldron/bw_matchbox/commit/0f243f885c597eefa3eda1ad28ea7f5899079618

**Comment by lilliputten** [comment]
Implemented in PR #144.

Viewed processes data stored in cookie named `recent-processes` (constants parameter `storeRecentProcessesCookieName`) in format:

```json
[{"id":28370,"time":1700559534730},{"id":28189,"time":1700559524486},{"id":27712,"time":1700558454588}]
```

This data updated on each process view. Stored only 20 last records, sorted in order of time (the recents are the first). See parameter `storeRecentProcessesCount` (it's possible to use `storeRecentProcessesForDays` parameter to specify time to store records.

Server api method `processes-attributes-json` used to extend stored data with process names.

![20231121-163833](https://github.com/cauldron/bw_matchbox/assets/6855837/a7b71add-ee9e-4dd8-aac5-d6e39f0ecddf)
